src/ray/gcs/gcs_server/gcs_placement_group_manager.cc

```diff
@@ -410,19 +410,21 @@ void GcsPlacementGroupManager::HandleWaitPlacementGroupUntilReady(
   RAY_LOG(DEBUG) << "Waiting for placement group until ready, placement group id = "
                  << placement_group_id;
 
-  // If the placement group does not exist or it has been successfully created, return
-  // directly.
+  // If the placement group has been successfully created, return directly.
   const auto &iter = registered_placement_groups_.find(placement_group_id);
-  if (iter == registered_placement_groups_.end()) {
-    RAY_LOG(DEBUG) << "Placement group is not exist, placement group id = "
-                   << placement_group_id;
-    GCS_RPC_SEND_REPLY(send_reply_callback, reply,
-                       Status::NotFound("Placement group is not exist."));
-  } else if (iter->second->GetState() == rpc::PlacementGroupTableData::CREATED) {
+  if (iter != registered_placement_groups_.end() &&
+      iter->second->GetState() == rpc::PlacementGroupTableData::CREATED) {
     RAY_LOG(DEBUG) << "Placement group is created, placement group id = "
                    << placement_group_id;
     GCS_RPC_SEND_REPLY(send_reply_callback, reply, Status::OK());
   } else {
+    // NOTE: We don't check if the placement group exist. There are two reasons:
+    // 1.`wait` is a method of placement group object. Placement group object is
+    // obtained by create placement group api, so it can guarantee the existence of
+    // placement group. Currently, we will not call the wait api after deleting placement
```

Comment:
`we will not call the wait api after deleting placement`

This sounds like a little dangerous assumption? It is still possible to do this right? 

```python3
pg = create_pg()
remove(pg)
pg.wait()
```
What's going to happen in this case?

Reply:
`wait` will return timeout, but the callback in GCS will not be deleted.

Reply:
so, in this case, the callback will always exist in gcs lifecycle?  can we check whether it should be deleted  in `HandleRemovePlacementGroup` method?

Reply:
@clay4444 `HandleRemovePlacementGroup` is called before `HandleWaitPlacementGroupUntilReady`.

Reply:
em...., yes, maybe we can get the pg state from redis and check whether it's state is `Removed`, but I think there are some overdesign?  and, I think the implements of ready api have same issue too? @rkooo567 ,

Reply:
ready API doesn't have the same issue because it doesn't call the GCS I think (and maybe as a proof, tests weren't flaky). If ready is broken, that means the placement group is fundamentally broken. 

@ffbin You said using a sequencer requires just a few lines of code right? Why don't we just go with that method? Wil there be any problem there?

Reply:
@rkooo567 A sequencer need to wait for the previous RPC result to return before calling the next request. In this case, the timeout of wait will not be accurate.

Reply:
@rkooo567 @clay4444 I plan to add a judgment in GCS. If the PG cannot be found, query it from the storage.

Reply:
I think in case that the PG is already deleted on "wait", the right semantic is to throw an app-level exception, instead of timing out.

Reply:
what @raulchen said sounds reasonable.

Reply:
Ah, @clay4444 you are right that the ready API also has the issue that if it is called after removal, it is troublesome. (I was thinking about different cases).

@ffbin I agree with @raulchen about the semantic. I think can ping Redis, and if it is already removed, we can throw an exception.

Reply:
SGTM, i have fix it, thanks.